apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -149,12 +150,14 @@ private void CheckCompatibleMode(TransactionMode mode)
             switch (mode)
             {
                 case TransactionMode.ReadOnly:
-                    Mode.AssertTrue(x => x == TransactionMode.ReadOnly || x == TransactionMode.ReadWrite,
-                        "You can only execute reads on a ReadWrite or ReadOnly Transaction!");
+                {
+                    GaxPreconditions.CheckState(((Predicate<TransactionMode>) (x => x == TransactionMode.ReadOnly || x == TransactionMode.ReadWrite))(Mode), "You can only execute reads on a ReadWrite or ReadOnly Transaction!");
```

Comment:
I think it would be much simpler to use `CheckState(Mode == TransactionMode.ReadOnly || Mode == TransactionMode.ReadWrite` - it's a read-only property, so it won't change between calls...